lib/elements/utils.py

```diff
@@ -49,6 +49,12 @@ def node_to_elements(node, clone_to_element=False):  # noqa: C901
             elements.reverse()
         return elements
 
+    elif node.tag == SVG_GROUP_TAG:
+        elements = []
+        for child in node.getchildren():
```

Comment:
Can you change this to `node.iterdescendants()` to save us some iterations for nested groups?

Reply:
I can, but then if I have a graph like this, iterdescendants would result in use1's clone of path2 being visited twice, once from `node_to_elements(g1)`, and once from `node_to_elements(g2)` as `path2` is a descendant of both `g1` and `g2`.  
![image](https://github.com/inkstitch/inkstitch/assets/5505503/3b9b26c0-a9a3-4d8f-9f63-eb11ffee5722)

Though I suppose this can be worked around e.g. by not calling `node_to_elements` on any non-embroiderable children. Is recursion considered that undesirable here?

Reply:
My first intuition was to rather not have the clone group logic in node_to_element. This proves it again. So let's leave the utils file untouched and move the logic into the clone file. This should work better and also takes care for the cache problem. It'd be a bad idea to clear the cache whenever a clone is rendered.

```
    def clone_to_element(self, node):
        from .utils import node_to_elements
        elements = []
        if node.tag in EMBROIDERABLE_TAGS:
            elements = node_to_elements(node, True)
        elif node.tag == SVG_GROUP_TAG:
            for child in node.iterdescendants():
                elements.extend(node_to_elements(child, True))
        return elements
```

Reply:
And maybe rename to `clone_to_elements` since this is returning a list of elements.